### DISCUSSION TITLE
Logging, and other debugging tools?

### DISCUSSION THREAD

**Discussion by mpounsett** [mastodon/mastodon#23865] [Q&A] [answered]
The "Configuring Your Environment" section of the documentation notes the `RAILS_LOG_LEVEL` defaults to `info`.  I don't see anything about a logging location, or master on/off.  I have not modified the logging level in my config file, and the log level documentation suggests the default should be rather noisy, however the `~mastodon/live/log/` directory on my instance contains only a single (empty) `.keep` file, and I don't see anything related to Mastodon in `/var/log/`.    Is there something I'm missing here?  Something broken?  

What other tools for debugging are there when trying to look into an issue with the server?  The troubleshooting section of the documentation is more of a FAQ for a few specific known issues, rather than direction on where and how to find details about the software's activity and potential issues.

I'm running Mastodon v4.0.2 on Ubuntu 20.04.5 LTS, as installed by the 1-click droplet setup on Digital Ocean.  I'm looking to upgrade to 4.1.0, but would really like to know where my logs and other troubleshooting tools are, first.

**Comment by mpounsett** [comment]
I would still very much like to understand what's going on with logging on my server.  Is Mastodon just not expected to log anything, or is there something broken?

Since asking the initial question, we've upgraded a couple of times. Currently on v4.1.2

**Comment by Gargron** [comment, marked as answer]
Mastodon is a 12-factor application and as such, logs to stdout. Log management is delegated to whatever is running Mastodon, systemd, Docker, Kubernetes, etc.

**Comment by mpounsett** [comment]
Aha!  Thanks!  A note about "logs to stderr" in the logging section of the documentation would help a ton here.
